next-hardhat/test/index.ts

```diff
@@ -17,26 +16,40 @@ describe('It compiles noir program code, receiving circuit bytes and abi object.
 
   it('Should generate valid proof for correct input', async () => {
     const input = { x: 1, y: 2 };
-    await noir.init();
-    const witness = await noir.generateWitness(input);
-    correctProof = await noir.generateProof(witness);
 
-    expect(correctProof instanceof Uint8Array).to.be.true;
-  });
+    const backend = new BarretenbergBackend(circuit);
+    // Initialize program
+    const program = new Noir(circuit, backend);
+    // Generate proof
+    const proof = await program.generateFinalProof(input);
 
-  it('Should verify valid proof for correct input', async () => {
-    const verification = await noir.verifyProof(correctProof);
-    expect(verification).to.be.true;
+    // Proof verification
+    const isValid = await program.verifyFinalProof(proof);
+    expect(isValid).to.be.true;
   });
 
+  // TODO(Ze): My understanding is that this test was previously using a global noir instance
+  // TODO: Why was this the case and is this more indicative of the real workflow? 
+  // it('Should verify valid proof for correct input', async () => {
+  //   const verification = await program.verifyFinalProof(correctProof);
+  //   expect(verification).to.be.true;
+  // });
+
   it('Should fail to generate valid proof for incorrect input', async () => {
     try {
       const input = { x: 1, y: 1 };
-      await noir.init();
-      const witness = await noir.generateWitness(input);
-      correctProof = await noir.generateProof(witness);
+      
+      const backend = new BarretenbergBackend(circuit);
+      // Initialize program
+      const program = new Noir(circuit, backend);
+      // Generate proof
+      await program.generateFinalProof(input);
     } catch (err) {
+
+      // TODO(Ze): Not sure how detailed we want this test to be
```

Comment:
TODO(Ze): this was just a nit -- before it was just checking for an error. .contains now checks for a specific error, though this test is now more reliant on the messages that noir_js gives.

Reply:
Much better like this, I was just being a lazy fuck back when I wrote this test 😄